### PR TITLE
Feat: add new tonconnect for multi accounts

### PR DESCRIPTION
**Pull request by vzhovnitsky**
- fixing connections and apps for multi-accounts
- adding new connect ui

### CODE REVIEW

File: app/engine/hooks/dapps/useSetTonconnectConnections.ts [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
-import { connectionsSelector } from "../../state/tonconnect";
 import { ConnectedAppConnection } from '../../tonconnect/types';
+import { connectionsMapAtom } from "../../state/tonconnect";
 
 export function useSetAppsConnectionsState() {
-    return useRecoilCallback(({ set }) => (updater: (prev: { [key: string]: ConnectedAppConnection[] }) => { [key: string]: ConnectedAppConnection[] }) => {
-        set(connectionsSelector, updater);
-    }, []);
+    const callback = useRecoilCallback(({ set }) => (
+        udater: (doc: { [key: string]: ConnectedAppConnection[] }) => { [x: string]: ConnectedAppConnection[] },
```
Comment (dvlkv) [inline app/engine/hooks/dapps/useSetTonconnectConnections.ts:7]: updater

File: app/engine/hooks/dapps/useTonConnectExtenstions.ts [hunk 2]
```diff
@@ -10,7 +11,38 @@ export type ConnectedApp = {
...
+  return [extensions || {}, setterUpdater];
+}
+
+type Updater = (doc: { [key: string]: ConnectedApp }) => { [x: string]: ConnectedApp };
+
+export function useSetTonConnectExtensions() {
+  const callback = useRecoilCallback(({ set }) => (udater: (doc: { [key: string]: ConnectedApp }) => { [x: string]: ConnectedApp }, address: string) => {
+    set(connectExtensionsMapAtom, (state) => {
+      const newState = { ...state };
+      newState[address] = udater(newState[address] || {});
```
Comment (dvlkv) [inline app/engine/hooks/dapps/useTonConnectExtenstions.ts:41]: updater

File: app/engine/state/tonconnect.ts [hunk 3]
```diff
@@ -151,50 +139,68 @@ function getPendingRequestsState(address?: string) {
...
-  const selected = getCurrentAddressNullable();
-  storePendingRequestsState(newValue, selected?.addressString);
-}
+export type SendTransactionRequestsMap = { [address: string]: SendTransactionRequest[] }
 
-const pendingRequestsState = atom({
+const pendingRequestsState = atom<SendTransactionRequestsMap>({
   key: 'tonconnect/pendingRequests/state',
-  default: getPendingRequestsForCurrent(),
+  default: (() => {
```
Comment (dvlkv) [inline app/engine/state/tonconnect.ts:150]: вызов обычной функции inline?

**Review by dvlkv — APPROVED** [review]

File: app/engine/state/tonconnect.ts (hunk 3 above)
Comment (vzhovnitsky) [reply app/engine/state/tonconnect.ts]: ✅

**Review by dvlkv — APPROVED** [review]

**Pull request merged**